module-4/backend/week_two.md

```diff
@@ -1,22 +1,35 @@
 ## Week Two - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
-
-Note: When you're done, submit a PR. 
-
 1. What's one difference between ES5 and ES6?
-2. What's the difference between asynchronous and synchronous JavaScript? 
+* ES6 has introduced the availability of classes in JavaScript, the syntax for defining variables with the use of let and const, as well as the syntax for defining functions using a fat arrow "=>".
+2. What's the difference between asynchronous and synchronous JavaScript?
+* Synchronous JavaScript is where functions can only run one function at a time. The next function must way for the prior function to complete before it can begin running.
+* Asynchronous JavaScript takes advantage of the browsers ability to run more than one function at a time. It would first fire off a method that would come on to the stack and then be sent to a Web API so that it is no longer on the stack and be able to continue to execute code/functions before the response of the first function is complete. This can be advantageous, but should be used in a limited way otherwise the benefits are lost.
 3. What are the four pillars of Object Oriented programming?
+* Abstraction: The idea of pulling out parts of code for reusability and to follow the DRY and SRP.
+* Encapsulation: The idea of grouping functions together base on the shared similar behavior and attributes.
+* Inheritance: The idea of passing on functionality to another class that can use and build upon the functionality provided by that class.
+* Polymorphism: The idea that a function can be used in a different way to execute similar behavior. For example, a Model T has an engine and a Corvette has an engine. Though both have engines there are parts of the engine that are different though the concept of the engine is similar.
 4. What are some tools available in JavaScript to help you write object oriented code?
+* Classes are a tool provided by JavaScript to help write more object oriented code.
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+* Being aware of repetition, functions that are too long, naming of variables and functions that are not clear, leaving in debuggers statements are all things to look for in code.
 6. What's a callback function and what are some reasons when we use/need callback functions?
+* Call back functions are functions that are passed to another function as a parameter. These are good to use in promises.
```

Comment:
Yep, and certain functions/methods specifically take callbacks as parameters (think back to array.map, etc, those were all taking callbacks!)